tests/Analyze/Gen.hs

```diff
@@ -0,0 +1,464 @@
+         mkBool $ DecimalComparison op (extract x) (extract y)
+  , do op <- genComparisonOp
+       Gen.subtermM2 (genCore BoundedTime) (genCore BoundedTime) $ \x y ->
+         mkBool $ TimeComparison op (extract x) (extract y)
+  , do op <- genComparisonOp
+       Gen.subtermM2 (genCore strSize) (genCore strSize) $ \x y ->
+         mkBool $ StringComparison op (extract x) (extract y)
+  , do op <- Gen.element [Eq, Neq]
+       Gen.subtermM2 (genCore BoundedBool) (genCore BoundedBool) $ \x y ->
+         mkBool $ BoolComparison op (extract x) (extract y)
+  , do op <- Gen.element [AndOp, OrOp]
+       Gen.subtermM2 (genCore BoundedBool) (genCore BoundedBool) $ \x y ->
+         mkBool $ Logical op [extract x, extract y]
+  , Gen.subtermM (genCore BoundedBool) $ \x ->
+      mkBool $ Logical NotOp [extract x]
+  ]
+genCore BoundedTime = Gen.recursive Gen.choice [
+    ESimple TTime . CoreTerm . Lit <$> Gen.enumBounded -- Gen.int64
+  ] $ scale 4 <$> [
+    Gen.subtermM2 (genCore BoundedTime) (genCore (BoundedInt 1e9)) $ \x y ->
+      pure $ ESimple TTime $ Inj $ IntAddTime (extract x) (extract y)
+  , Gen.subtermM2 (genCore BoundedTime) (genCore (BoundedDecimal 1e9)) $ \x y ->
+      pure $ ESimple TTime $ Inj $ DecAddTime (extract x) (extract y)
+  ]
+genCore BoundedKeySet = ESimple TKeySet . CoreTerm . Lit . KeySet
+  <$> genInteger (0 ... 2)
+
+intSize, decSize, strSize :: BoundedType
+intSize = BoundedInt     (0 +/- 1e25)
+decSize = BoundedDecimal (0 +/- 1e25)
+strSize = BoundedString  1000
+
+-- TODO
+-- generic:
+-- # Let
+-- object:
+-- # Read
+-- string:
+-- # Write
+-- time:
+-- # ParseTime
+genAnyTerm
+  :: (MonadGen m, MonadReader GenEnv m, MonadState GenState m, HasCallStack)
+  => m ETerm
+genAnyTerm = Gen.choice
+  [ genTerm intSize
+  , genTerm decSize
+  , genTerm strSize
+  , genTerm BoundedBool
+  , genTerm BoundedTime
+  -- , genTerm BoundedKeySet
+  ]
+
+genTerm
+  :: (MonadGen m, MonadReader GenEnv m, MonadState GenState m, HasCallStack)
+  => BoundedType -> m ETerm
+genTerm size = scale 2 $ Gen.choice [genCore size, genTermSpecific size]
+
+genTermSpecific
+  :: (MonadGen m, MonadReader GenEnv m, MonadState GenState m, HasCallStack)
+  => BoundedType -> m ETerm
+genTermSpecific size@BoundedInt{} = genTermSpecific' size
+genTermSpecific BoundedBool       = Gen.choice
+  [
+  -- TODO: temporarily disabled pending
+  -- https://github.com/kadena-io/pact/issues/207
+  -- [ ESimple TBool . Enforce (Just 0) . extract <$> genTerm BoundedBool
+  -- , do xs <- Gen.list (Range.linear 0 4) (genTerm BoundedBool)
+  --      pure $ ESimple TBool $ EnforceOne $ case xs of
+  --        [] -> Left 0
+  --        _  -> Right $ fmap (((Path 0, Path 0),) . extract) xs
+
+  -- TODO(joel):
+  -- , do tagId <- genTagId
+  --      ESimple TBool . KsAuthorized tagId . extract <$> genTerm BoundedKeySet
+  -- , do tagId <- genTagId
+  --      ESimple TBool . NameAuthorized tagId . extract <$> genTerm strSize
```

Comment:
any blockers here?

Reply:
No -- also things we don't cover yet. I'll update the comments with more information.